### ISSUE TITLE
LSP returns strings with `\\\n` at the end of lines

### ISSUE THREAD

**Issue by michelesr** [microsoft/pylance-release#3139]
**Describe the bug**

I'm using pyright with the neovim LSP built in client. When I look at hover documentation (or completion items documentation), I can see trailing backslash at the end of some lines (see screenshot)

[
![screenshot-2022-08-03-16:35:39](https://user-images.githubusercontent.com/2516077/182649365-b97eaa84-f97a-4c2a-b35a-7e983b82d0d5.png)
](url)

Inspecting the responses from the LSP server I can see this:

```
[DEBUG][2022-08-03 16:36:44] .../vim/lsp/rpc.lua:454    "rpc.receive"   {  id = 81,  jsonrpc = "2.0",  result = {    contents = {      kind = "markdown",      value = "```python\n(function) abstractmethod: (funcobj: _FuncT@abstractmethod) -> _FuncT@abstractmethod\n```\n---\nA decorator indicating abstract methods.\n\nRequires that the metaclass is ABCMeta or derived from it.  A\nclass that has a metaclass derived from ABCMeta cannot be\ninstantiated unless all of its abstract methods are overridden.\nThe abstract methods can be called using any of the normal\n'super' call mechanisms.  abstractmethod() may be used to declare\nabstract methods for properties and descriptors.\n\nUsage:\n\n&nbsp;&nbsp;&nbsp;&nbsp;class C(metaclass=ABCMeta):\\\n&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@abstractmethod\ndef my\\_abstract\\_method(self, ...):\\\n&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;..."    },    range = {      end = {        character = 18,        line = 2      },      start = {        character = 4,        line = 2      }    }  }}
```

In that string there are instances of `\\\n` at the end of the line.

**To Reproduce**
I don't know how to reproduce in other editor/configurations. In neovim with configured lspconfig is sufficient to trigger the hover function

**Expected behavior**
There shouldn't be a `\\\n` in the end of line (not sure about things like `&nbps;`). It should probably just be `\n`

**Screenshots or Code**
```
import abc

# hover on abstractmethod
abc.abstractmethod
```

**VS Code extension or command-line**
pyright version 1.1.264

**Additional context**
A workaround is mentioned in a [reddit thread](https://old.reddit.com/r/neovim/comments/tx40m2/is_it_possible_to_improve_lsp_hover_look/) but the problem is not only present when doing an hover (for example it can be reproduced also on documentation associated with a completion item)

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.8.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022821-10-august-2022-prerelease)

**Comment by michelesr** [comment]
I'm not sure I follow: I thought this was an issue in the pyright language server? How is it fixed there? There's nothing in the changelog about this issue.

**Comment by bschnurr** [comment]
this was the pyright PR.  usually changes from pylance get squashed together to update pyright. this PR was missing detailed info though.
https://github.com/microsoft/pyright/pull/3816

**Comment by erictraut** [comment]
This will be included in the next release of pyright (1.1.267), which should be published in the next two days.
